### PR TITLE
Simplify the DoPoolForward/Backward

### PR DESCRIPTION
**Pull request by kaixih**
This PR simplifies the code base for the pooling ops by following the style set by the `DoConvolve()`. This can also help reducing the duplicate code for our ongoing PR that supports big tensors.

cc. @nluehr

### CODE REVIEW

File: tensorflow/stream_executor/cuda/cuda_dnn.cc [hunk 1]
```diff
@@ -6009,194 +6009,82 @@ bool CudnnSupport::DoActivate(Stream* stream,
...
-    DeviceMemory<Eigen::half>* output_data,
+port::Status CudnnSupport::DoPoolForward(
+    dnn::DataType element_type, Stream* stream,
+    const dnn::PoolingDescriptor& pooling_dimensions,
+    const dnn::BatchDescriptor& input_dimensions, DeviceMemoryBase input_data,
+    const dnn::BatchDescriptor& output_dimensions, DeviceMemoryBase output_data,
     ScratchAllocator* workspace_allocator) {
   // Alpha is the scaling factor for input.
-  float alpha = 1.0;
+  float alpha_f = 1.0f;
```
Comment (jurahul) [inline tensorflow/stream_executor/cuda/cuda_dnn.cc:6019]: Can we mark these const and the pointers const as well?

File: tensorflow/stream_executor/cuda/cuda_dnn.cc [hunk 2]
```diff
@@ -6009,194 +6009,82 @@ bool CudnnSupport::DoActivate(Stream* stream,
...
+    dnn::DataType element_type, Stream* stream,
+    const dnn::PoolingDescriptor& pooling_dimensions,
+    const dnn::BatchDescriptor& input_dimensions, DeviceMemoryBase input_data,
+    const dnn::BatchDescriptor& output_dimensions, DeviceMemoryBase output_data,
+    DeviceMemoryBase input_diff_data, DeviceMemoryBase output_diff_data,
     ScratchAllocator* workspace_allocator) {
   // Alpha is the scaling factor for input.
-  double alpha = 1.0;
+  float alpha_f = 1.0f;
+  double alpha_d = 1.0;
```
Comment (jurahul) [inline tensorflow/stream_executor/cuda/cuda_dnn.cc:6058]: Same comment as above. Please add const.

Comment (kaixih) [reply tensorflow/stream_executor/cuda/cuda_dnn.cc]: Sure. I've pushed a new commit. PTAL.

**Comment by kaixih** [comment]
Gentle ping: Any updates? @jurahul

**Review by jurahul — APPROVED** [review]

**Review by mihaimaruseac — APPROVED** [review]

**Comment by kaixih** [comment]
"feedback/copybara" failed but I don't have access to the log. Can you take a look? @jurahul  Thx.

**Comment by jurahul** [comment]
Yes, I am looking into this. There are build failures from this change in internal code, which we need to fix on our side before this can be merged.

**Pull request merged**